File: 0027-remove-element/0027-remove-element.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        vector<int> ret;
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] == val) {
                continue;
            } else {
                ret.push_back(nums[i]);
            }
        }
        nums = ret;

        return nums.size();
    }
};

File: 0027-remove-element/0027-remove-element.go
func removeElement(nums []int, val int) int {
    
    j := 0
    for i := 0; i < len(nums); i++ {
        if nums[i] != val {
            nums[j], nums[i] = nums[i], nums[j]
            j++
        }
    }

    return j
}